spec/rubocop/cop/rails/root_join_chain_spec.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Rails::RootJoinChain, :config do
+  it 'does not register an offense for `Rails.root.join(...)`' do
+    expect_no_offenses(<<~RUBY)
+      Rails.root.join('db', 'schema.rb')
+    RUBY
+  end
+
+  it 'registers and offense and corrects for `::Rails.root.join(...).join(...)`' do
+    expect_offense(<<~RUBY)
+      ::Rails.root.join('db').join('sch' + 'ema.rb')
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `Rails.root.join(...)` instead of chaining `#join` calls.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      ::Rails.root.join('db', 'sch' + 'ema.rb')
+    RUBY
+  end
+
+  it 'registers and offense and corrects for `Rails.root.join(...).join(...)`' do
+    expect_offense(<<~RUBY)
+      Rails.root.join('db').join('sch' + 'ema.rb')
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `Rails.root.join(...)` instead of chaining `#join` calls.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      Rails.root.join('db', 'sch' + 'ema.rb')
+    RUBY
+  end
+
+  it 'registers and offense and corrects for `Rails.root` with any number of joins greater one' do
+    expect_offense(<<~RUBY)
+      Rails.root.join.join.join('db').join(migrate).join.join("migration.\#{rb}")
+      ^^^^^^^^^^^^^^^^^^^^ Use `Rails.root.join(...)` instead of chaining `#join` calls.
```

Comment:
It looks like all the `join` methods in the chain should be highlighted.

```suggestion
      Rails.root.join.join.join('db').join(migrate).join.join("migration.\#{rb}")
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `Rails.root.join(...)` instead of chaining `#join` calls.
```